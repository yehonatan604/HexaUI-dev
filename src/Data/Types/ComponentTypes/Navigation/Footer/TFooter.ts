import { TComponent, ComponentOptions } from "../../TComponent";

export type TFooter = TComponent<"div"> & {
    options?: FooterOptions;
    reserved?: string;
}

export type FooterOptions = ComponentOptions & {
    reservedVariant?: string;
    childrenJustify?: string;
    childrenAlign?: string;
    childrenDirection?: string;
}