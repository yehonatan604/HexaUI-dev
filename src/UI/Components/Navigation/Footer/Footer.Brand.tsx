import { TBrand } from "../../../../Data/Types/TBrand";

const FooterBrand = (props: TBrand) => {
  const { children, as, to, ...componentProps } = props;
  const Component = as || "a";

  return (
    <Component className="text-xl font-bold" {...componentProps} href={to} to={to}>
      {children}
    </Component>
  );
};

export default FooterBrand;
