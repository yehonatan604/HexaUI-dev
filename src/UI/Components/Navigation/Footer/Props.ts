import { TComponent } from "../../../../Data/Types/TComponent";

export type FooterProps = TComponent<"div"> & {
    reserved?: string;
    reservedColor?: string;
    bgColor?: string;
    textColor?: string;
    childrenJustify?: string;
    childrenAlign?: string;
    childrenDirection?: string;
};