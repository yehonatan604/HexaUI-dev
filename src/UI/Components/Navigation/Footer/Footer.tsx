import { FlexDir } from "../../../../Data/Constants/FlexDirection";
import { FlexTypes } from "../../../../Data/Constants/FlexTypes";
import { TFooter } from "./types/TFooter";
import Flex from "../../Layout/Flex/Flex";
import FooterBrand from "./Footer.Brand";

const Footer = (props: TFooter) => {
  const {
    children,
    className,
    reserved,
    options: {
      bgVariant,
      textVariant,
      reservedVariant,
      childrenDirection,
      childrenJustify,
      childrenAlign,
    },
    ...componenetProps
  } = props;

  // Options
  const bgColor = bgVariant || "white";
  const textColor = textVariant || "black";
  const reservedColor = reservedVariant || "standard";
  const childrenGap = childrenDirection === FlexDir.Row ? "gap-4" : "gap-1";

  return (
    <Flex
      options={{
        direction: FlexDir.Col,
        justify: FlexTypes.Start,
        align: FlexTypes.Start,
      }}
      className={`bg-${bgColor} text-${textColor} ${className} `}
      {...componenetProps}
    >
      <Flex
        options={{
          direction: childrenDirection,
          justify: childrenJustify,
          align: childrenAlign,
        }}
        className={`w-full ${childrenGap} max-md:flex-col`}
      >
        {children}
      </Flex>
      <p className={`text-center text-xs text-${reservedColor} max-md:my-2`}>
        &copy; {new Date().getFullYear()} {reserved || "Rainbow UI"}
      </p>
    </Flex>
  );
};

Footer.Brand = FooterBrand;
export default Footer;
