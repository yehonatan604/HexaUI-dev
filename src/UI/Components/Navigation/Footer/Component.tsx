import { FlexDir } from "../../../../Data/Constants/FlexDirection";
import { FlexTypes } from "../../../../Data/Constants/FlexTypes";
import Flex from "../../Layout/Flex/Component";
import FooterBrand from "./Footer.Brand/Component";
import { FooterProps } from "./Props";

const Footer = (props: FooterProps) => {
  const {
    children,
    className,
    reserved,
    bgColor,
    textColor,
    reservedColor,
    childrenJustify = FlexTypes.Center,
    childrenAlign = FlexTypes.Center,
    childrenDirection = FlexDir.Row,
    ...componenetProps
  } = props;

  const childrenGap = childrenDirection === FlexDir.Row ? "gap-4" : "gap-1";

  return (
    <Flex
      direction={FlexDir.Col}
      justify={FlexTypes.Center}
      align={FlexTypes.Center}
      className={`${className} ${bgColor} ${textColor}`}
      {...componenetProps}
    >
      <Flex
        direction={childrenDirection}
        justify={childrenJustify}
        align={childrenAlign}
        className={`w-full ${childrenGap} max-md:flex-col`}
      >
        {children}
      </Flex>
      <p className={`text-center text-xs ${reservedColor} max-md:my-2`}>
        &copy; {new Date().getFullYear()} {reserved || "Rainbow UI"}
      </p>
    </Flex>
  );
};

Footer.Brand = FooterBrand;
export default Footer;
