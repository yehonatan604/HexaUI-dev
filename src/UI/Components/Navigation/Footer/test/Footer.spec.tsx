import { render, screen } from "@testing-library/react";
import Footer from "../Footer"; // Adjust the import path
import { describe, it, expect } from "vitest";

describe("Footer Component", () => {
  it("should render children correctly", () => {
    render(
      <Footer options={{ childrenDirection: "row" }}>
        <div>Child 1</div>
        <div>Child 2</div>
      </Footer>
    );

    // Check if both children are rendered
    expect(screen.getByText("Child 1")).toBeInTheDocument();
    expect(screen.getByText("Child 2")).toBeInTheDocument();
  });

  it("should display the correct reserved text", () => {
    render(
      <Footer reserved="My Company" options={{}}>
        Footer Content
      </Footer>
    );

    // Check if the reserved text is displayed correctly
    const currentYear = new Date().getFullYear();
    expect(screen.getByText(`© ${currentYear} My Company`)).toBeInTheDocument();
  });

  it("should display default reserved text when not provided", () => {
    render(<Footer options={{}}>Footer Content</Footer>);

    // Check if the default reserved text is displayed
    const currentYear = new Date().getFullYear();
    expect(screen.getByText(`© ${currentYear} Rainbow UI`)).toBeInTheDocument();
  });

  it("should apply custom background and text colors", () => {
    render(
      <Footer options={{ bgVariant: "blue-500", textVariant: "white" }}>
        Footer Content
      </Footer>
    );

    const footerElement = screen.getByText("Footer Content").parentElement;

    // Check if the custom background and text colors are applied
    expect(footerElement).toHaveClass("bg-blue-500", "text-white");
  });

  it("should apply the correct reserved text variant", () => {
    render(<Footer options={{ reservedVariant: "gray-500" }}>Footer Content</Footer>);

    // Check if the correct reserved text variant is applied
    expect(screen.getByText(/©/)).toHaveClass("text-gray-500");
  });

  it("should render and match the snapshot", () => {
    const { asFragment } = render(
      <Footer
        options={{
          bgVariant: "blue-500",
          textVariant: "white",
          reservedVariant: "gray-500",
          childrenDirection: "row",
        }}
      >
        <div>Snapshot Child</div>
      </Footer>
    );

    // Snapshot the Footer structure
    expect(asFragment()).toMatchSnapshot();
  });
});
