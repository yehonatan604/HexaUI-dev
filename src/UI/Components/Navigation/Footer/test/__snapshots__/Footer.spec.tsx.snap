// Vitest Snapshot v1

exports[`Footer Component > should render and match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col bg- text- justify-start items-start flex-nowrap bg-blue-500 text-white undefined "
  >
    <div
      class="flex flex-row bg- text- justify-start items-start flex-nowrap w-full gap-4 max-md:flex-col"
    >
      <div>
        Snapshot Child
      </div>
    </div>
    <p
      class="text-center text-xs text-gray-500 max-md:my-2"
    >
      Â© 2025 Rainbow UI
    </p>
  </div>
</DocumentFragment>
`;
